terminus-ui/logo/src/logo.component.spec.ts

```diff
@@ -0,0 +1,175 @@
+import { Component, ViewChild } from '@angular/core';
+import {
+  ComponentFixture,
+  TestBed,
+  TestModuleMetadata,
+} from '@angular/core/testing';
+import { configureTestBedWithoutReset } from '@terminus/ngx-tools/testing';
+import { By } from '@angular/platform-browser';
+
+import { TsLogoComponent } from './logo.component';
+import { TsLogoModule } from './logo.module';
+
+
+@Component({
+  template: `
+    <ts-logo id="one" #one></ts-logo>
+    <ts-logo id="two" #two [type]="type" [logoColor]="logoColor"></ts-logo>
+  `,
+})
+class TestHostComponent {
+  type = '';
+  logoColor = '';
+
+  @ViewChild('one')
+  one!: TsLogoComponent;
+
+  @ViewChild('two')
+  two!: TsLogoComponent;
+}
+
+
+
+describe(`TsLogoComponent`, function() {
+  let fixture: ComponentFixture<TestHostComponent>;
+  let hostComponent: TestHostComponent;
+  let logo1: TsLogoComponent;
+  let logo2: TsLogoComponent;
+  const moduleDefinition: TestModuleMetadata = {
+    imports: [
+      TsLogoModule,
+    ],
+    declarations: [
+      TestHostComponent,
+    ],
+  };
+
+  configureTestBedWithoutReset(moduleDefinition);
+
+  beforeEach(() => {
+    // Reset parent component inputs
+    if (hostComponent) {
+      hostComponent.type = '';
+      fixture.detectChanges();
+    }
+
+    fixture = TestBed.createComponent(TestHostComponent);
+    hostComponent = fixture.componentInstance;
+    logo1 = hostComponent.one;
+    logo2 = hostComponent.two;
+  });
+
+
+  test(`should exist`, () => {
+    expect(logo1).toBeTruthy();
+    expect(logo2).toBeTruthy();
+  });
+
+
+  describe(`logo type`, () => {
+
+    test(`should default to full-gradient without input`, () => {
+      expect(logo1.type).toEqual('full-gradient');
+    });
+
+    test(`should log a warning if an invalid value was passed in`, () => {
+      window.console.warn = jest.fn();
+      hostComponent.type = 'foo';
+      fixture.detectChanges();
+
+      expect(window.console.warn).toHaveBeenCalled();
+    });
+
+    test(`should change type to full-account-hub on input`, () => {
+
+      hostComponent.type = 'full-account-hub';
+      fixture.detectChanges();
+      expect(logo2.type).toEqual('full-account-hub');
```

Comment:
We should probably be checking that the actual SVG output is what we expect here rather than just the TS type. We can add an ID inside the SVG code to make it easier to determine which SVG is loaded if it helps

Reply:
Great idea!